amulet/deployer.py

```diff
@@ -136,6 +136,15 @@ def add(self, service_name,
         if placement is not None:
             service['to'] = placement
 
+        if constraints is None and 'JUJU_TEST_CONSTRAINTS' in os.environ:
```

Comment:
Yes, I get that, but there's no way to verify that. Trust but verify. End of the day, if this satisfies @dannf's needs I'm good with it. If we ever come up to a point where we need to override authors constraints a `JUJU_TEST_CONSTRAINTS_OVERRIDES` flag, or similar, could be added